test/com/google/enterprise/adaptor/database/JdbcFixture.java

```diff
@@ -71,7 +123,22 @@ public static void dropAllObjects() throws SQLException {
         throw new AssertionError(e);
       }
     }
-    executeUpdate("drop all objects");
+    if (database.equalsIgnoreCase("h2")) {
+      executeUpdate("drop all objects");
+    } else if (database.equalsIgnoreCase("sqlserver")) {
+      dropSQLServerTables();
+      // TODO (srinivas): drop tables for Oracle database.
+    }
+  }
+
+  private static void dropSQLServerTables() throws SQLException {
+    String sql = "select 'DROP TABLE ' + name + '' FROM sys.tables";
```

Comment:
This is confusing. Just "select name from sys.tables" here and do the string concatenation below. Are you sure that doesn't delete anything we don't want to delete?